data_model/src/config.rs

```diff
@@ -0,0 +1,296 @@
+    }
+
+    /// Log level for reading from environment and (de)serializing
+    #[allow(clippy::upper_case_acronyms)]
+    #[derive(Debug, Deserialize, Serialize, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+    pub enum Level {
+        /// Error
+        ERROR,
+        /// Warn
+        WARN,
+        /// Info (Default)
+        INFO,
+        /// Debug
+        DEBUG,
+        /// Trace
+        TRACE,
+    }
+}
+
+/// Module for genesis-related configuration and structs.
+pub mod genesis {
+    use super::*;
+
+    #[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+    #[serde(rename_all = "UPPERCASE")]
+    /// Configuration of the genesis block and the process of its submission.
+    pub struct Configuration {
+        /// The genesis account public key, should be supplied to all peers.
+        pub account_public_key: PublicKey,
+        /// Number of attempts to connect to peers, while waiting for them to submit genesis.
+        pub wait_for_peers_retry_count_limit: u64,
+        /// Period in milliseconds in which to retry connecting to peers, while waiting for them to submit genesis.
+        pub wait_for_peers_retry_period_ms: u64,
+        /// Delay before genesis block submission after minimum number of peers were discovered to be online.
+        /// Used to ensure that other peers had time to connect to each other.
+        pub genesis_submission_delay_ms: u64,
+    }
+}
+
+/// Module for wsv-related configuration and structs.
+pub mod wsv {
+    use super::*;
+    use crate::{metadata::Limits as MetadataLimits, LengthLimits};
+
+    /// `WorldStateView` configuration.
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize)]
+    #[serde(rename_all = "UPPERCASE")]
+    pub struct Configuration {
+        /// [`MetadataLimits`] for every asset with store.
+        pub asset_metadata_limits: MetadataLimits,
+        /// [`MetadataLimits`] of any asset definition's metadata.
+        pub asset_definition_metadata_limits: MetadataLimits,
+        /// [`MetadataLimits`] of any account's metadata.
+        pub account_metadata_limits: MetadataLimits,
+        /// [`MetadataLimits`] of any domain's metadata.
+        pub domain_metadata_limits: MetadataLimits,
+        /// [`LengthLimits`] for the number of chars in identifiers that can be stored in the WSV.
+        pub ident_length_limits: LengthLimits,
+        /// WASM Runtime configuration
+        pub wasm_runtime_config: wasm::Configuration,
+    }
+}
+
+/// Module for wasm-related configuration and structs.
+pub mod wasm {
+    use super::*;
+    /// `WebAssembly Runtime` configuration.
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize)]
+    #[serde(rename_all = "UPPERCASE")]
+    pub struct Configuration {
+        /// Every WASM instruction costs approximately 1 unit of fuel. See
+        /// [`wasmtime` reference](https://docs.rs/wasmtime/0.29.0/wasmtime/struct.Store.html#method.add_fuel)
```

Comment:
@outoftardis, maybe something like this will be better?

> Maximum amount of fuel provided to smart contract.
> Limits the number of instructions that can be executed during the execution of a smart contract. 
> Every WASM instruction costs approximately 1 unit of fuel.
> See [`wasmtime` reference](https://docs.rs/wasmtime/0.29.0/wasmtime/struct.Store.html#method.add_fuel).

Reply:
much better, yes. 
you can make the first two sentences more concise, like this:

> The fuel limit determines the maximum number of instructions that can be executed during the execution of a smart contract.